src/main/java/io/cdap/e2e/utils/WaitHelper.java

```diff
@@ -267,4 +267,20 @@ public static boolean waitForTextToBePresentInElementValue(WebElement element, S
                   " with the Default timeout: " + ConstantsUtil.DEFAULT_TIMEOUT_SECONDS + " seconds");
     return SeleniumDriver.getWaitDriver().until(ExpectedConditions.textToBePresentInElementValue(element, text));
   }
+
+  /**
+   * Wait for the element to be disabled
+   *
+   * @param element WebElement to check whether click is disabled for temporary time
+   */
+  public static boolean waitForElementToBeDisabled(WebElement element) {
+    logger.info("Waiting for the element: " + element + " to be disabled ");
+    try {
+      SeleniumDriver.getWaitDriver().until(ExpectedConditions.elementToBeClickable(element));
```

Comment:
I don't think this is semantically correct based on what the function is expected to do

Reply:
We have changed the method contents as discussed , please see the below file .
 public static boolean waitForElementToBeOptionallyDisabled(WebElement element) {
    logger.info("The element: " + element + " is disabled ");
    return !element.isEnabled();